rllib/evaluation/rollout_worker.py

```diff
@@ -1519,18 +1546,31 @@ def _determine_spaces_for_multi_agent_dict(
     return multi_agent_dict
 
 
-def _validate_env(env: Any) -> EnvType:
-    # Allow this as a special case (assumed gym.Env).
-    if hasattr(env, "observation_space") and hasattr(env, "action_space"):
-        return env
+def _validate_env(env: EnvType, env_context: EnvContext = None):
+
+    msg = "Validating sub-env at vector index=0 ... "
 
     allowed_types = [
         gym.Env, MultiAgentEnv, ExternalEnv, VectorEnv, BaseEnv,
         ray.actor.ActorHandle
     ]
     if not any(isinstance(env, tpe) for tpe in allowed_types):
-        raise ValueError(
+        raise EnvError(
             "Returned env should be an instance of gym.Env, MultiAgentEnv, "
             "ExternalEnv, VectorEnv, or BaseEnv. The provided env creator "
-            "function returned {} ({}).".format(env, type(env)))
-    return env
+            "function returned {} (type={}).".format(env, type(env)))
+
+    # Do some test runs with the provided env.
+    if isinstance(env, gym.Env):
+        assert hasattr(env, "observation_space") and hasattr(
+            env, "action_space")
+        dummy_obs = env.reset()
+        if not env.observation_space.contains(dummy_obs):
+            msg += "(NOT OK)"
+            logger.warning(msg)
+            raise EnvError(
+                f"Env's `observation_space` {env.observation_space} does not "
+                f"contain returned observation after a reset ({dummy_obs})!")
+        elif env_context.vector_index == 0:
```

Comment:
can you add a comment here about why we are checking vector_index? looks a bit confusing to newbies like me.

actually a minor nit picking suggestion is that maybe this will read a little better if we re-structure the checks like:

```
if not env.observation_space.contains(dummy_obs):
  raise EnvError(...)
elif env_context.vector_index != 0:
  raise Error(...some kind of err...)

# Everything is good if we are here.
msg += "(ok)"
logger.info(msg)
```

that way, all the errors will raise an Error before ok condition. and we can easily extract this into a more complicated module later as we discussed offline.
also, I guess as written right now, if vector_index != 0, we don't print anything or raise any errors.

Reply:
will do!

Reply:
Actually, the point here is to not INFO-log this for every single worker (if we have 30+ workers, this would become quite spammy). But still error-check (and throw the error) for all workers.

Reply:
Ah, sorry, you are right. Confusing vector index with worker index :/ 
Will fix this.

Reply:
But same argument. Only INFO-logging here for vector-index=0 so not to spam the logs.
I cleaned up the block, made it more clear and added more comments. :)